ide/rust/RepoCargoManifestGenerator.kt

```diff
@@ -0,0 +1,325 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.dependencies.ide.rust
+
+import com.electronwill.nightconfig.core.Config
+import com.electronwill.nightconfig.toml.TomlWriter
+import com.vaticle.bazel.distribution.common.shell.Shell
+import com.vaticle.bazel.distribution.common.util.FileUtil.listFilesRecursively
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Type.BIN
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Type.BUILD
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Type.LIB
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Type.TEST
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.Paths.BAZEL_BIN
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.Paths.CARGO_TOML
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.Paths.EXTERNAL
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.Paths.EXTERNAL_PLACEHOLDER
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.Paths.IDE_SYNC_PROPERTIES
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.ShellArgs.BAZEL
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.ShellArgs.INFO
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.BUILD_DEPS
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.DEPS_PREFIX
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.EDITION
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.ENTRY_POINT_PATH
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.FEATURES
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.NAME
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.PATH
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.ROOT_PATH
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.SOURCES_ARE_GENERATED
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.TYPE
+import com.vaticle.dependencies.ide.rust.RepoCargoManifestGenerator.TargetSyncInfo.Keys.VERSION
+import java.io.File
+import java.io.FileInputStream
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.Properties
+import kotlin.io.path.Path
+
+class RepoCargoManifestGenerator(private val workspaceRoot: File, shell: Shell) {
```

Comment:
For consistency, the field `workspaceRoot` should be named `repository`, don't you think?